File: plugins/setup/src/main/java/net/elytrarace/setup/conversation/map/MapPrompt.java
package net.elytrarace.setup.conversation.map;

import net.elytrarace.api.conversation.BooleanPrompt;
import net.elytrarace.api.conversation.ConversationContext;
import net.elytrarace.api.conversation.Prompt;
import net.kyori.adventure.text.Component;
import org.jetbrains.annotations.NotNull;
import org.jetbrains.annotations.Nullable;

public class MapPrompt extends BooleanPrompt {
    @Override
    public @NotNull Component getPromptText(@NotNull ConversationContext context) {
        return Component.translatable("prompt.map");
    }

    @Override
    protected @Nullable Prompt acceptValidatedInput(@NotNull ConversationContext context, boolean input) {
        if (!input) {
            return END_OF_CONVERSATION;
        }
        return new MapCupNamePrompt();
    }
}

File: src/main/kotlin/net/elytrarace/conversation/map/MapPrompt.kt
package net.elytrarace.conversation.map

import net.elytrarace.conversation.BooleanPrompt
import net.elytrarace.conversation.ConversationContext
import net.elytrarace.conversation.Prompt
import net.kyori.adventure.text.Component


class MapPrompt : BooleanPrompt() {
    override fun getPromptText(context: ConversationContext): Component {
        return Component.translatable("prompt.world")
    }

    override fun acceptValidatedInput(context: ConversationContext, input: Boolean): Prompt? {
        if (!input) {
            return END_OF_CONVERSATION
        }
        return MapCupNamePrompt()
    }
}